rpcs3/CMakeLists.txt

```diff
@@ -160,11 +160,18 @@ if(NOT LLVM_FOUND)
 else()
 	add_definitions(${LLVM_DEFINITIONS})
 	add_definitions(-DLLVM_AVAILABLE)
-	if (CMAKE_BUILD_TYPE STREQUAL "Release")
-		llvm_map_components_to_libnames(LLVM_LIBS mcjit vectorize ipo x86codegen x86disassembler)
+	
+	#Use static linking for windows binary but always dynamic on *nix
+	if (WIN32 OR MSVC)
+		if (CMAKE_BUILD_TYPE STREQUAL "Release")
+			llvm_map_components_to_libnames(LLVM_LIBS mcjit vectorize ipo x86codegen x86disassembler)
+		else()
+			llvm_map_components_to_libnames(LLVM_LIBS mcjit vectorize ipo x86codegen x86disassembler mcdisassembler)
+		endif()
 	else()
-		llvm_map_components_to_libnames(LLVM_LIBS mcjit vectorize ipo x86codegen x86disassembler mcdisassembler)
+		set(LLVM_LIBS LLVM-4.0.so)
```

Comment:
I haven't found a single option that links to the .so instead of the .a modules. Most of these options seem to affect compiling the llvm runtime rather than affecting the target linkage. I'll keep checking, but I suspect the only option we'll eventually have is to change the components_list